src/cider/nrepl/middleware/inspect.clj

```diff
@@ -48,47 +53,79 @@
 
 (defn pop-reply
   [{:keys [transport] :as msg}]
-  (let [inspector (swap-inspector! msg inspect/up)]
-    (transport/send
-     transport
-     (response-for msg :value (:rendered inspector) :status :done))))
+  (try
```

Comment:
This pattern is used so many times we should refactoring it into a macro. Something like `with-exception-as-error-messages`.